content/inside-rust/stage0-std-redesign.md

```diff
@@ -0,0 +1,53 @@
++++
+title = "Redesigning stage 0 std"
+authors = ["Jieyou Xu"]
+
+[extra]
+team = "Bootstrap"
+team_url = "https://www.rust-lang.org/governance/teams/infra#team-bootstrap"
++++
+
+# Redesigning stage 0 std
+
+## Summary
+
+We are reworking how the stage 0 bootstrap sequence works (the sequence used to build a stage 1 compiler).
+
+- Before: a stage 0 (possibly beta) compiler is used to build the in-tree std, which in turn is used to build the stage 1 compiler.
+- After: a stage 0 (possibly beta) compiler and a precompiled stage 0 std is instead used to build the stage 1 compiler.
+
+Notably, this means:
+
+- `./x {build,test,check} library --stage 0` becomes no-op, as stage 0 std is no longer the built in-tree std, and the minimum supported stage to build std is now `1`.
+    - Consequently, default (test, check, bench) stage values in the library profile are no longer `0`.
+- `cfg(bootstrap)` is no longer needed for library development. In (very) rare cases, `cfg(bootstrap)` may be needed in the compiler for dogfooding unstable library features.
```

Comment:
> In (very) rare cases, `cfg(bootstrap)` may be needed in the compiler for dogfooding unstable library features.

Note that we already use cfg(bootstrap) on compiler tree for some cases, so cfg(bootstrap) usage on compiler tree won't be a new thing.